be/src/vec/runtime/vparquet_writer.cpp

```diff
@@ -0,0 +1,782 @@
+            parquet_data_type = parquet::Type::INT32;
+        } else if (data_type.find("int64") != std::string::npos) {
+            parquet_data_type = parquet::Type::INT64;
+        } else if (data_type.find("int96") != std::string::npos) {
+            parquet_data_type = parquet::Type::INT96;
+        } else if (data_type.find("float") != std::string::npos) {
+            parquet_data_type = parquet::Type::FLOAT;
+        } else if (data_type.find("double") != std::string::npos) {
+            parquet_data_type = parquet::Type::DOUBLE;
+        } else if (data_type.find("byte_array") != std::string::npos) {
+            parquet_data_type = parquet::Type::BYTE_ARRAY;
+        } else if (data_type.find("fixed_len_byte_array") != std::string::npos) {
+            parquet_data_type = parquet::Type::FIXED_LEN_BYTE_ARRAY;
+        } else {
+            parquet_data_type = parquet::Type::UNDEFINED;
+        }
+
+        std::string column_name = (*column)[2];
+        fields.push_back(parquet::schema::PrimitiveNode::Make(column_name, parquet_repetition_type,
+                                                              parquet::LogicalType::None(),
+                                                              parquet_data_type));
+        _schema = std::static_pointer_cast<parquet::schema::GroupNode>(
+                parquet::schema::GroupNode::Make("schema", parquet::Repetition::REQUIRED, fields));
+    }
+    return Status::OK();
+}
+
+Status VParquetWriterWrapper::write(const Block& block) {
+    if (block.rows() == 0) {
+        return Status::OK();
+    }
+    size_t sz = block.rows();
+    try {
+        for (size_t i = 0; i < block.columns(); i++) {
+            auto& col = block.get_by_position(i).column;
+            auto& type = block.get_by_position(i).type;
+            switch (_output_vexpr_ctxs[i]->root()->type().type) {
+            case TYPE_BOOLEAN: {
+                if (_str_schema[i][1] != "boolean") {
+                    return Status::InvalidArgument(
+                            "project field type is boolean, "
+                            "but the definition type of column {} is {}",
+                            _str_schema[i][2], _str_schema[i][1]);
+                }
+                parquet::RowGroupWriter* rgWriter = get_rg_writer();
+                parquet::BoolWriter* col_writer =
+                        static_cast<parquet::BoolWriter*>(rgWriter->column(i));
+                bool default_bool = false;
+                if (const auto* not_nullable_column =
+                            check_and_get_column<const ColumnVector<UInt8>>(col.get())) {
+                    col_writer->WriteBatch(
+                            sz, nullptr, nullptr,
+                            reinterpret_cast<const bool*>(not_nullable_column->get_data().data()));
+                } else if (const auto* nullable_column =
+                                   check_and_get_column<const ColumnNullable>(col.get())) {
```

Comment:
here check nullable column, it maybe placed on the top of switch, so that all cases not need to write check nullable code